code/lib/cli/src/automigrate/fixes/webpack5-compiler-setup.ts

```diff
@@ -0,0 +1,208 @@
+        (addon) =>
+          addon.includes(CoreWebpackCompilers.Babel) || addon.includes(CoreWebpackCompilers.SWC)
+      )
+    ) {
+      return null;
+    }
+
+    const frameworkName = Object.entries(frameworkPackages).find(
+      ([name]) => name === getFrameworkPackageName(mainConfig)
+    )?.[1];
+
+    const builderPackageName = getBuilderPackageName(mainConfig);
+    const customCoreBuilder = builderPackageName
+      ? builderNameToCoreBuilder[builderPackageName]
+      : null;
+
+    const defaultCoreBuilder = frameworkName
+      ? frameworkToDefaultBuilder[frameworkName]
+      : await (async () => {
+          const webpackVersion = await packageManager.getPackageVersion('webpack');
+          return !!webpackVersion ? CoreBuilder.Webpack5 : CoreBuilder.Vite;
+        })();
+
+    const builder = customCoreBuilder || defaultCoreBuilder;
+
+    if (builder !== CoreBuilder.Webpack5) {
+      return null;
+    }
+
+    const excludedFrameworks: SupportedFrameworks[] = ['angular', 'ember'];
+
+    const isExcludedFramework = frameworkName ? excludedFrameworks.includes(frameworkName) : false;
+
+    if (isExcludedFramework) {
+      return null;
+    }
+
+    const hasReactScriptsDependency = !!(await packageManager.getPackageVersion('react-scripts'));
+
+    if (hasReactScriptsDependency) {
+      return null;
+    }
+
+    const frameworkOptions = getFrameworkOptions(mainConfig);
+
+    const defaultCompiler = frameworkOptions?.builder?.useSWC
+      ? CoreWebpackCompilers.SWC
+      : CoreWebpackCompilers.Babel;
+
+    const shouldRemoveSWCFlag = frameworkOptions?.builder
+      ? 'useSWC' in frameworkOptions.builder
+      : false;
+
+    if (frameworkName === 'nextjs') {
+      return {
+        compiler: undefined,
+        compilerPackageName: undefined,
+        shouldRemoveSWCFlag,
+        isNextJs: true,
+      };
+    }
+
+    return {
+      defaultCompiler,
+      shouldRemoveSWCFlag,
+      isNextJs: false,
+    };
+  },
+
+  prompt({ defaultCompiler, shouldRemoveSWCFlag, isNextJs }) {
+    const message = [];
+
+    if (shouldRemoveSWCFlag) {
+      message.push(dedent`
+      We need to update your Storybook configuration for Webpack 5.
+      The ${chalk.yellow('framework.options.builder.useSWC')} flag will be removed.`);
+    }
+
+    if (isNextJs) {
+      message.push(dedent`
+      Storybook now detects whether it should use Babel or SWC as a compiler by applying the same logic as Next.js itself:\n
+        - If you have a ${chalk.yellow(
+          'babel.config.js'
+        )} file in your project, Storybook will use Babel as the compiler.
+        - If you have a ${chalk.yellow('babel.config.js')} file in your project and you have set 
+          ${chalk.yellow('experimental.forceSwcTransforms = true')} in your ${chalk.yellow(
+            'next.config.js'
+          )} file, 
+          Storybook will use SWC as the compiler.
+        - If you don't have a ${chalk.yellow(
+          'babel.config.js'
```

Comment:
Is it only `babel.config.js` or are there other babel config types supported (.babelrc / babel.config.json)?

Reply:
You are right, also `.babelrc` is supported (https://nextjs.org/docs/pages/building-your-application/configuring/babel)